# 106-110年在台僑生及港澳生人數 

In [3]:
import pandas as pd
data = pd.read_csv('overseasstudent.csv')
data.head()

學年度 設立別  學校類別 學校統計處代碼    學校名稱      系所代碼         系所名稱 學制班別(日間)  僑生小計  僑生男  \
0  106  公立  一般大學    0001  國立政治大學  01111001         教育學系  學士班(日間)    19    2   
1  106  公立  一般大學    0001  國立政治大學  01111001         教育學系  碩士班(日間)     3    2   
2  106  公立  一般大學    0001  國立政治大學  01114005   教育行政與政策研究所  碩士班(日間)     1    1   
3  106  公立  一般大學    0001  國立政治大學  01131002  華語文教學博士學位學程      博士班     1    0   
4  106  公立  一般大學    0001  國立政治大學  01131002  華語文教學碩士學位學程  碩士班(日間)     2    0   

   僑生女  港澳生小計  港澳生男  港澳生女  僑生及港澳生數之在學比率(%)  
0   17      5     3     2            10.04  
1    1      0     0     0             6.52  
2    0      1     0     1             4.26  
3    1      0     0     0             5.26  
4    2      1     0     1             5.26

# 1.106－110學年合計有多少名僑生在台留學?

In [4]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['僑生小計']]
total = df_selected.sum().sum()

In [5]:
print(f'106-110學年合計有{total}名僑生在台留學')

106-110學年合計有61967名僑生在台留學


# 2-4. 當中男生佔多少人？女生佔多少人？比率為?

In [6]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['僑生男']]
total_male = df_selected.sum().sum()

In [7]:
print(f'106-110學年合計有{total_male}名男僑生在台留學')

106-110學年合計有29655名男僑生在台留學


In [8]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['僑生女']]
total_female = df_selected.sum().sum()

In [9]:
print(f'106-110學年合計有{total_female}名女僑生在台留學')

106-110學年合計有32312名女僑生在台留學


In [10]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['僑生男']]
total_male = df_selected.sum().sum()

df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['僑生女']]
total_female = df_selected.sum().sum()

In [11]:
ratio = total_male / total_female

In [12]:
print("僑生男女比率為：", ratio)

僑生男女比率為： 0.9177704877444912


# 5.106－110學年合計有多少名港澳生在台留學?

In [13]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['港澳生小計']]
total = df_selected.sum().sum()

In [14]:
print(f'106-110學年合計有{total}名港澳生在台留學')

106-110學年合計有58298名港澳生在台留學


# 6-8.當中男生佔多少人？女生佔多少人？比率為?

In [15]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['港澳生男']]
total_male = df_selected.sum().sum()

In [16]:
print(f'106-110學年合計有{total_male}名男港澳生在台留學')

106-110學年合計有32096名男港澳生在台留學


In [17]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['港澳生女']]
total_female = df_selected.sum().sum()

In [18]:
print(f'106-110學年合計有{total_female}名女港澳生在台留學')

106-110學年合計有26202名女港澳生在台留學


In [19]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['港澳生男']]
total_male = df_selected.sum().sum()

df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['港澳生女']]
total_female = df_selected.sum().sum()

In [20]:
ratio = total_male / total_female

In [21]:
print("港澳生男女比率為：", ratio)

港澳生男女比率為： 1.2249446607129226


# 9. 106－110學年合計有多少名僑生及港澳生在台留學?

In [22]:
df = pd.read_csv('overseasstudent.csv', encoding='utf-8')
df_selected = df[['僑生小計','港澳生小計']]
total = df_selected.sum().sum()

In [23]:
print(f'106-110學年合計有{total}名僑生及港澳生在台留學')

106-110學年合計有120265名僑生及港澳生在台留學


# 10. 有人說就讀公立大學的僑生、港澳生比就讀私立大學的多，你認同嗎？試就資料解釋。

我不認同，以下為就讀公立大學的僑生及港澳生的總數。

In [24]:
df_public = df.loc[df['設立別'] == '公立']
total_count = df_public['僑生小計'].sum() + df_public['港澳生小計'].sum()
print("就讀公立大學的學生總人數：", total_count)

就讀公立大學的學生總人數： 59292


以下為就讀私立大學的僑生及港澳生的總數。

In [26]:
df_private = df.loc[df['設立別'] == '私立']
total_count = df_private['僑生小計'].sum() + df_private['港澳生小計'].sum()
print("就讀私立大學的學生總人數：", total_count)

就讀私立大學的學生總人數： 60973


由此可見，就讀私立大學的僑生及港澳生比就讀公立大學的約多出1700人，可能的原因是台灣的私立大學數量比公立大學多，而且大部分私立大學的收生門檻比公立大學低，因此有此情況出現。

# 11. 試找出106－110學年各年在台就學的僑生及港澳生數量，並比較當中的差異。

以下為106學年在台僑生及港澳生數量。

In [31]:
df_106 = df.loc[df['學年度'] == 106]
total_count = df_106['僑生男'].sum() + df_106['僑生女'].sum() + df_106['港澳生女'].sum() + df_106['港澳生男'].sum()
print("106學年在台的僑生及港澳生總人數：", total_count)

106學年在台的僑生及港澳生總人數： 24594


以下為107學年在台僑生及港澳生數量。

In [32]:
df_107 = df.loc[df['學年度'] == 107]
total_count = df_107['僑生男'].sum() + df_107['僑生女'].sum() + df_107['港澳生女'].sum() + df_107['港澳生男'].sum()
print("107學年在台的僑生及港澳生總人數：", total_count)

107學年在台的僑生及港澳生總人數： 23850


以下為108學年在台僑生及港澳生數量。

In [33]:
df_108 = df.loc[df['學年度'] == 108]
total_count = df_108['僑生男'].sum() + df_108['僑生女'].sum() + df_108['港澳生女'].sum() + df_108['港澳生男'].sum()
print("108學年在台的僑生及港澳生總人數：", total_count)

108學年在台的僑生及港澳生總人數： 22651


以下為109學年在台僑生及港澳生數量。

In [34]:
df_109 = df.loc[df['學年度'] == 109]
total_count = df_109['僑生男'].sum() + df_109['僑生女'].sum() + df_109['港澳生女'].sum() + df_109['港澳生男'].sum()
print("109學年在台的僑生及港澳生總人數：", total_count)

109學年在台的僑生及港澳生總人數： 23516


以下為110學年在台僑生及港澳生數量。

In [35]:
df_110 = df.loc[df['學年度'] == 110]
total_count = df_110['僑生男'].sum() + df_110['僑生女'].sum() + df_110['港澳生女'].sum() + df_110['港澳生男'].sum()
print("110學年在台的僑生及港澳生總人數：", total_count)

110學年在台的僑生及港澳生總人數： 25654


由此可見，108－109學年的僑生及港澳生人數比106－107學年少，估計是受Covid－19影響，而到110學年因疫情放緩，吸引更多僑生及港澳生來台就學，因此該年在台的僑生及港澳生總人數為過去5年的最多，達25654人，比109年多出2138人。